src/main/java/org/opentripplanner/transit/raptor/api/path/Path.java

```diff
@@ -43,13 +45,23 @@ private Path(int startTime, int endTime, int numberOfTransfers, int generalizedC
         this.egressPathLeg = null;
     }
 
-    public Path(AccessPathLeg<T> accessLeg, int generalizedCost) {
-        this.accessLeg = accessLeg;
-        this.egressPathLeg = findEgressLeg(accessLeg);
+    public Path(int iteration, AccessPathLeg<T> accessLeg, int generalizedCost) {
+        this.iteration = iteration;
         this.startTime = accessLeg.fromTime();
-        this.endTime = egressPathLeg.toTime();
         this.numberOfTransfers = countNumberOfTransfers(accessLeg);
         this.generalizedCost = generalizedCost;
+        this.accessLeg = accessLeg;
+        this.egressPathLeg = findEgressLeg(accessLeg);
+        this.endTime = egressPathLeg.toTime();
+    }
+
+    /**
+     * The Range Raptor iteration. This can be used in the path-pareto-function to make sure
+     * all results found in previous iterations are kept, and not dominated by new results.
```

Comment:
Clarify Javadoc: "all results found in previous iterations are kept" at the egress stations, but not in the main search itself, where the range raptor optimization depends on kicking out states from previous (later departure) iterations.